dbms/src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -217,6 +217,7 @@ MergeTreeData::DataPart::Checksums checkDataPart(
     MergeTreeData::DataPart::Checksums checksums_data;
 
     size_t marks_in_primary_key = 0;
+    if (Poco::File(path + "primary.idx").exists())
```

Comment:
Looks like incorrect change. What if primary key must exist but the file was removed.

Why don't just move `if (!primary_key_data_types.empty())` upper?

Reply:
Ok.
`primary_key_data_types` can be empty to skip check of the primary key.
But it's needed only for `part_checker` tool. I'd better remove `part_checker` tool.